internal/controller/k0smotron.io/k0smotroncluster_statefulset.go

```diff
@@ -146,6 +147,19 @@ func (r *ClusterReconciler) generateStatefulSet(kmc *km.Cluster) (apps.StatefulS
 		})
 	}
 
+	if kmc.Spec.Manifests != "" {
+		statefulSet.Spec.Template.Spec.Volumes = append(statefulSet.Spec.Template.Spec.Volumes, v1.Volume{
+			Name: "manifests",
+			VolumeSource: v1.VolumeSource{
+				ConfigMap: &v1.ConfigMapVolumeSource{
+					LocalObjectReference: v1.LocalObjectReference{Name: kmc.Spec.Manifests},
+				}}})
+
+		statefulSet.Spec.Template.Spec.Containers[0].VolumeMounts = append(statefulSet.Spec.Template.Spec.Containers[0].VolumeMounts, v1.VolumeMount{
+			Name:      "manifests",
+			MountPath: "/var/lib/k0s/manifests/custom",
```

Comment:
What if instead adding a single `custom` "stack" (in k0s implementation terms) we iterate over all the keys and add each key as a separate stack?

Reply:
There is certainly a benefit of having more than one stack for better error handling, I hadn't thought of that...  However, doing that would require the operator to watch for changes and redeploy the stateful set in the event of changes of the configmap.

What do you think of allowing multiple secrets instead of just one? I mean changing the type of `Manifests` from `string` to `[]string`. It would allow multiple stacks, and if there are changes in the `Cluster`object we already monitor for that.

Reply:
> I mean changing the type of Manifests from string to []string. It would allow multiple stacks, and if there are changes in the Clusterobject we already monitor for that.

That makes sense.

Although I started to wonder if we can achieve this with plain secrets. I know you can mount different values to different files pretty automatically, but can the key contain a path? Probably worth of some trialing. I mean if that would work, then we wouldn't have to even re-create the statefullset pods as k8s will sync the secret mount automatically, IIRC.

Reply:
> I mean changing the type of Manifests from string to []string

Makes a lot of sense. In theory, different manifests may have different sources.

Reply:
> Although I started to wonder if we can achieve this with plain secrets. I know you can mount different values to different files pretty automatically, but can the key contain a path?

As per kubernetes documentation no:
The keys of data and stringData must consist of alphanumeric characters, `-`, `_` or `.`.

I don't think doing this in a single secret is feasible without using subPath volume mounts but that would require us to monitor the secrets and I don't think this is something we want...

Reply:
👌